### ISSUE TITLE
Invalid params for paramstyles based on position, not names

### ISSUE THREAD

**Issue by giaume** [antonrh/jinjaq#14]
Reading https://peps.python.org/pep-0249/#paramstyle and https://peps.python.org/pep-0249/#id53, my understanding is that parameter styles fall into two different categories:

1. `named`, `pyformat` where parameters are referenced by their **name**, and `execute()` second argument is typically a dict
2. `qmark`, `numeric`, `format` where parameters are referenced by their **position**, and `execute()` second argument is typically a list

Jinja2SQL works great when you use a database engine using a parameter style of the first category, but when using one of the second category:

- the 2nd element of the tuple returned by `from_string()` can't be used directly as 2nd parameter for `execute()`
- a workaround could be to use `.values()` but it doesn't work for obvious reasons when a parameter is mentioned more than once in the query

```
from jinja2sql import Jinja2SQL

j2sql = Jinja2SQL(param_style="format")
query, params = j2sql.from_string(
    "SELECT * FROM users WHERE username = {{ user }} OR email = {{ user }}",
    context={"user": "user@example.com"},
)
# query: "SELECT * FROM users WHERE username = %s OR email = %s"
# params: {'user': 'user@example.com'}
# list(params.values()): ['user@example.com']

conn.execute(query, params) # fails because params isn't a sequence
conn.execute(query, list(params.values()) # fails because we have "%s" twice in the query, but 'user@example.com' only once in the parameter list
```

**Comment by antonrh** [comment]
Hey, @giaume , Thank you for pointing out this issue. The problem should now be resolved. Previously, I was trying to use a custom parameter type that worked for both positional and keyword parameters. Now, I’ve changed it to a union of list and dict, since most dbapi implementations use these types. The only adjustment needed is to cast it to a dict when using it with SQLAlchemy's `.bindparams` — for example, `sa.select(query).bindparams(**dict(params))`.

**Comment by antonrh** [comment]
Fix available in [0.6.0](https://pypi.org/project/jinja2sql/)
